source/extensions/quic_listeners/quiche/codec_impl.h

```diff
@@ -0,0 +1,58 @@
+#include "envoy/http/codec.h"
+
+#include "common/common/assert.h"
+#include "common/common/logger.h"
+
+#include "extensions/quic_listeners/quiche/envoy_quic_server_session.h"
+
+namespace Envoy {
+namespace Quic {
+
+// QuicHttpConnectionImplBase instance is a thin QUIC codec just providing quic interface to HCM.
+// Owned by HCM and created during onNewConnection() if the network connection
+// is a QUIC connection.
+class QuicHttpConnectionImplBase : public virtual Http::Connection,
+                                   protected Logger::Loggable<Logger::Id::quic> {
+public:
+  QuicHttpConnectionImplBase(EnvoyQuicServerSession& quic_session) : quic_session_(quic_session) {}
+
+  // Http::Connection
+  void dispatch(Buffer::Instance& /*data*/) override {
+    // Bypassed. QUIC connection already hands all data to streams.
+    NOT_REACHED_GCOVR_EXCL_LINE;
```

Comment:
sadly these don't quite work, and so add to our uncovered lines.
It really bites when we get close to 97% coverage because if you add some NOT_REACHED_GCOVR_EXCL_LINE in a small CL you then have to write unit tests for unrelated code.

Let's instead add quiche to COVERAGE_IGNORE_REGEX in test/run_envoy_bazel_coverage.sh and have a tracking item in one of the QUICHE bugs to fix up coverage before we land.
I would like you to take a look at coverge results and see if any of the new code not explicitly NOT_REACHED makes sense to unit test, because I suspect much of it does and it's easier to land in line than do it all at once.  If you don't know how to dig through the CI -> coverage -> artifacts I can run you through it.